pkg/datapath/tables/route.go

```diff
@@ -82,6 +82,22 @@ func (r *Route) String() string {
 		r.Dst, r.Src, r.Table, r.LinkIndex)
 }
 
+func (*Route) TabHeader() string {
```

Comment:
Not sure if there might be other use cases for this but perhaps we could have something more generic or perhaps even auto generated via struct tags?? Seems single purpose in this form imho. ie Perhaps something like Table interface with header []string and row map[string]interface{} might be more flexible, for cols ordering/filtering, tooling or if custom output is desired or query perf becomes an issue for larger tables and computed fields??

Reply:
Hmm, I wouldn't auto-generate this, but would rather hand-write so that the output looks good and makes sense. The default serialization for some values is often not what we want and it's fair bit of boilerplate to suddenly needing to overwrite e.g. `String()` on things. Good example is `netip.Addr` showing "invalid IP" for zero value.

Single-purpose and simple is good here IMO, at least at this early point. And there's always `json.Marshal` and so on, and for filtering we can pipe via `jq`.

Reply:
Ok tried `TabRow() []string` and it seems little bit annoying with having to make many `Sprintf` calls:
```
 func (d *Device) TabRow() []string {
         addrs := []string{}
         for _, addr := range d.Addrs {
                 addrs = append(addrs, addr.Addr.String())
         }
         return []string{
                 d.Name,
                 fmt.Sprintf("%s", d.Index),
                 fmt.Sprintf("%v", d.Selected),
                 d.Type,
                 fmt.Sprintf("%d", d.MTU),
                 d.HardwareAddr.String(),
                 d.Flags.String(),
                 strings.Join(addrs, ", "),
         }
 }
```

versus:
```
func (d *Device) TabRow() string { 
         b := &strings.Builder{}
         fmt.Fprintf(b, "%s\t%d\t%v\t%s\t%d\t%s\t%s\t",
                 d.Name, d.Index, d.Selected,
                 d.Type, d.MTU, d.HardwareAddr, d.Flags)
         addrs := []string{}
         for _, addr := range d.Addrs {
                 addrs = append(addrs, addr.Addr.String())
         }
         fmt.Fprintf(b, "%s\n", strings.Join(addrs, ", "))
         return b.String()
 }
```

OTOH, the "%s\t%s\t..." soup is worse! I'll change it to `TabRow() []string` :D

Reply:
Could templates be of help here? So you can go
```
func (d *Device) TabRow() string { 
  tpl := "{{.Name}}\t{{.Index}}\t{{.Selected}}\t{{.Type}}\t{{.MTU}}\t{{.HardwareAddr}}\t{{.Flags}}\t{{join .Addrs ', ' }}"
  t, err := template.New("device").Funcs(template.FuncMap{"join": strings.Join}).Parse(tpl)
  if err != nil {
    // TODO: handle error
  }
  var buf bytes.Buffer
  err = t.Execute(&buf, d)
  if err != nil {
    // TODO: handle error
  }
  return buf.String()
}
```

Perhaps we can make a helper func out of the template parsing + execution so you don't have to repeat it all the time, but the template is at least clear.